app/src/main/java/org/hackillinois/android/viewmodel/AdminViewModel.kt

```diff
@@ -0,0 +1,85 @@
+package org.hackillinois.android.viewmodel
+
+import android.R
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import okhttp3.ResponseBody
+import org.hackillinois.android.App
+import org.hackillinois.android.model.*
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class AdminViewModel : ViewModel() {
+    var stats = MutableLiveData<String>()
+    var eventCreated = MutableLiveData<Boolean>()
+    var notificationTopics = MutableLiveData<NotificationTopics>()
+    var notificationCreated = MutableLiveData<Boolean>()
+
+    fun queryStats() {
+        App.getAPI().stats.enqueue(object: Callback<ResponseBody> {
+            override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
+                stats.postValue(null)
+            }
+
+            override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
+                response.body()?.let {
+                    val data = JSONObject(it.string()).toString(4)
+                    stats.postValue(data)
+                }
+            }
+        })
+    }
+
+    fun createEvent(name: String, description: String, sponsor: String, eventType: String,
+                    eventRoom: String, startTime: Long, endTime: Long, locationName: String) {
+        var location: EventLocation? = null
+        when(locationName) {
+            "Siebel Center" -> location = EventLocation(SiebelCenter.description + " " + eventRoom, SiebelCenter.latitude, SiebelCenter.longitude)

```

Comment:
You can do string concatenation like this as well in Kotlin:
`"${SiebelCenter.description} $eventRoom"`